### PR TITLE
Select the last tab and update selection after deleting the current tab

### PR DESCRIPTION
**Pull request by 0nko**
Task/Issue URL: https://app.asana.com/0/1202552961248957/1202148076467684/f

### Description
This PR selects the last tab in the list after the currently-selected tab is deleted. This fixes the behaviour when no tab was selected and when the user went back to the browser, the first tab was shown.

### Steps to test this PR

1. Create at least 3 tabs (with the last tab being selected)
2. Delete the last (selected) tab
3. Notice the tab is removed and the last of the remaining tab is selected
4. Tap on `Undo`
5. Notice the previous tab is shown and selected again
6. Delete the selected tab again
7. Tap on the back button
8. Notice the correct tab is shown

### UI changes
[Screen_recording_20240626_151442.webm](https://github.com/duckduckgo/Android/assets/1522856/2ae2fe33-275b-4eb8-a33c-62b647715d09)

### CODE REVIEW

File: app/src/main/java/com/duckduckgo/app/tabs/ui/TabGridItemDecorator.kt [hunk 1]
```diff
@@ -86,7 +86,7 @@ class TabGridItemDecorator(
     }
 
     companion object {
-        private val BORDER_RADIUS = 9.toPx().toFloat()
+        private val BORDER_RADIUS = 15.toPx().toFloat()
```
Comment (0nko) [inline app/src/main/java/com/duckduckgo/app/tabs/ui/TabGridItemDecorator.kt:89]: This is not directly related to the issue, but after working with the decorator I noticed the radius doesn’t match the card view (see the PR description).

**Review by anikiki — APPROVED** [review]
Looks good and works as expected! :tada:
